pennylane_lightning/lightning_qubit/_measurements.py

```diff
@@ -285,27 +287,37 @@ def measure_final_state(self, circuit: QuantumScript) -> Result:
     # pylint:disable = too-many-arguments
     def measure_with_samples(
         self,
-        mps: List[Union[SampleMeasurement, ClassicalShadowMP, ShadowExpvalMP]],
+        measurements: List[Union[SampleMeasurement, ClassicalShadowMP, ShadowExpvalMP]],
         shots: Shots,
+        mid_measurements=None,
     ) -> List[TensorLike]:
         """
         Returns the samples of the measurement process performed on the given state.
         This function assumes that the user-defined wire labels in the measurement process
         have already been mapped to integer wires used in the device.
 
         Args:
-            mps (List[Union[SampleMeasurement, ClassicalShadowMP, ShadowExpvalMP]]):
+            measurements (List[Union[SampleMeasurement, ClassicalShadowMP, ShadowExpvalMP]]):
                 The sample measurements to perform
             shots (Shots): The number of samples to take
+            mid_measurements (None, dict): Dictionary of mid-circuit measurements
 
         Returns:
             List[TensorLike[Any]]: Sample measurement results
         """
+        # last N measurements are sampling MCMs in ``dynamic_one_shot`` execution mode
+        mps = measurements[0 : -len(mid_measurements)] if mid_measurements else measurements
+        skip_measure = (
+            any(v == -1 for v in mid_measurements.values()) if mid_measurements else False
+        )
 
         groups, indices = _group_measurements(mps)
 
         all_res = []
         for group in groups:
+            if skip_measure:
+                all_res.extend([None] * len(group))
+                continue
```

Comment:
what is this logic for?

Reply:
If any of the MCM values is `-1`, the state might not be normalized which may lead to errors in measurement processes. We therefore just skip it.